src/main/java/net/glowstone/command/minecraft/CloneCommand.java

```diff
@@ -0,0 +1,338 @@
+                    return false;
+                }
+                break;
+
+            default:
+                sender.sendMessage(ChatColor.RED + "Usage: " + usageMessage);
+                return false;
+        }
+
+        boolean overlaps = between(lowCorner.getBlockX(), highCorner.getBlockX(), to.getBlockX()) ||
+                           between(lowCorner.getBlockY(), highCorner.getBlockY(), to.getBlockY()) ||
+                           between(lowCorner.getBlockZ(), highCorner.getBlockZ(), to.getBlockZ());
+
+        if (overlaps && !cloneMode.isAllowedToOverlap()) {
+            sender.sendMessage(ChatColor.RED + "Usage: " + usageMessage);
+            return false;
+        }
+
+        int blocksCloned = 0;
+
+        int widthX = highCorner.getBlockX() - lowCorner.getBlockX() + 1;
+        int widthY = highCorner.getBlockY() - lowCorner.getBlockY() + 1;
+        int widthZ = highCorner.getBlockZ() - lowCorner.getBlockZ() + 1;
+
+        Iterable<Integer> xIter = to.getBlockX() < lowCorner.getBlockX() ? new ForwardsAxisIterable(widthX) : new BackwardsAxisIterable(widthX);
+        Iterable<Integer> yIter = to.getBlockY() < lowCorner.getBlockY() ? new ForwardsAxisIterable(widthY) : new BackwardsAxisIterable(widthY);
+        Iterable<Integer> zIter = to.getBlockZ() < lowCorner.getBlockZ() ? new ForwardsAxisIterable(widthZ) : new BackwardsAxisIterable(widthZ);
+
+        for (int x : xIter) {
+            for (int y : yIter) {
+                for (int z : zIter) {
+                    GlowBlock fromBlock = world.getBlockAt(lowCorner.getBlockX() + x, lowCorner.getBlockY() + y, lowCorner.getBlockZ() + z);
+
+                    if (blockFilter.shouldClone(fromBlock)) {
+                        GlowBlock toBlock = world.getBlockAt(to.getBlockX() + x, to.getBlockY() + y, to.getBlockZ() + z);
+                        toBlock.setTypeIdAndData(fromBlock.getTypeId(), fromBlock.getData(), false);
+
+                        BlockEntity fromEntity = fromBlock.getBlockEntity();
+                        if (fromEntity != null) {
+                            BlockEntity toEntity = toBlock.getChunk().createEntity(toBlock.getX(), toBlock.getY(), toBlock.getZ(), toBlock.getTypeId());
+                            if (toEntity != null) {
+                                CompoundTag entityTag = new CompoundTag();
+                                fromEntity.saveNbt(entityTag);
+                                toEntity.loadNbt(entityTag);
+                            }
+                        }
+
+                        if (cloneMode == CloneMode.MOVE) {
+                            fromBlock.setType(Material.AIR, false);
+                        }
+
+                        blocksCloned++;
+                    }
+                }
+            }
+        }
+
+        sender.sendMessage("Cloned " + blocksCloned + " blocks.");
```

Comment:
If `blocksCloned` ends up being 0, should another message pop up here? (e.g. No blocks were cloned)